CMakeLists.txt

```diff
@@ -179,24 +179,70 @@ endif ()
 
 # Reasonable default. CheckCompiler.m4 checks thoroughly for HP-UX's sake.
 set (MONGOC_SOCKET_ARG2 "struct sockaddr")
+set (MONGOC_SOCKET_ARG3 "socklen_t")
+
+if (NOT WIN32)
+
+set(MONGOC_ACCEPT_ARG2_LIST "struct sockaddr" "void")
+set(MONGOC_ACCEPT_ARG3_LIST "socklen_t" "size_t" "int")
+
+list(LENGTH MONGOC_ACCEPT_ARG2_LIST MONGOC_ACCEPT_ARG2_LIST_LEN)
+math(EXPR MONGOC_ACCEPT_ARG2_LIST_LEN_R "${MONGOC_ACCEPT_ARG2_LIST_LEN} - 1")
+list(LENGTH MONGOC_ACCEPT_ARG3_LIST MONGOC_ACCEPT_ARG3_LIST_LEN)
+math(EXPR MONGOC_ACCEPT_ARG3_LIST_LEN_R "${MONGOC_ACCEPT_ARG3_LIST_LEN} - 1")
+
+foreach(MONGOC_ACCEPT_ARG2_L RANGE ${MONGOC_ACCEPT_ARG2_LIST_LEN_R})
+foreach(MONGOC_ACCEPT_ARG3_L RANGE ${MONGOC_ACCEPT_ARG3_LIST_LEN_R})
```

Comment:
why not just?:

```
foreach (MONGOC_ACCEPT_ARG2_VAL in "struct sockaddr" "void")
foreach (MONGOC_ACCEPT_ARG3_VAL in "socklen_t" "size_t" "int")
```
Why do we need an indexed loop instead of looping over the values directly?